core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -216,13 +264,22 @@ private InstanceData performRefresh() throws InterruptedException, ExecutionExce
                   .toString()));
 
       synchronized (instanceDataGuard) {
-        currentInstanceData = Futures.immediateFuture(data);
-        nextInstanceData =
-            executor.schedule(this::performRefresh, secondsToRefresh, TimeUnit.SECONDS);
         // Refresh completed successfully, reset forceRefreshRunning.
         forceRefreshRunning = false;
+        currentRefreshFailure = null;
+        currentInstanceData = Futures.immediateFuture(data);
+
+        // Now update nextInstanceData to perform a refresh after the
+        // scheduled delay
+        ListenableFuture scheduleDelay =
+            executor.schedule(() -> {}, secondsToRefresh, TimeUnit.SECONDS);
```

Comment:
Is there any other ways to do this? On the surface it looks unusual.

Reply:
I found a better method call for this.